nio_cli/commands/buildspec.py

```diff
@@ -0,0 +1,96 @@
+import json, os, sys
+from nio.util.discovery import is_class_discoverable as _is_class_discoverable
+from niocore.core.loader.discover import Discover
+from .base import Base
+
+
+def is_class_discoverable(_class, default_discoverability=True):
+    return _is_class_discoverable(_class, default_discoverability)
+
+
+class BuildSpec(Base):
+
+    def __init__(self, options, *args, **kwargs):
+        super().__init__(options, *args, **kwargs)
+        self._repo = self.options['<repo-name>']
+
+    def run(self):
+        spec = {}
+        from nio.block.base import Block
+        sys.path.insert(0, os.getcwd())
+        blocks = Discover.discover_classes(
+            'blocks.{}'.format(self._repo), Block, is_class_discoverable)
+        for block in blocks:
+            k, v = self._build_spec_for_block(block)
+            spec[k] = v
+        file_path = 'blocks/{}/spec.json'.format(self._repo)
+        previous_spec = self._read_spec(file_path)
+        merged_spec = self._merge_previous_and_new_spec(previous_spec, spec)
+        with open(file_path, 'w') as f:
+            json.dump(merged_spec, f, sort_keys=True, indent=2)
+
+    def _read_spec(self, file_path):
+        if os.path.exists(file_path):
+            with open(file_path) as f:
+                return json.load(f)
+        else:
+            return {}
+
+    def _merge_previous_and_new_spec(self, previous_spec, spec):
+        # Merge in manually entered parts of previous spec
+        for block in spec:
+            spec[block]["Description"] = \
```

Comment:
These first lines are sort of merge/init. If you have a prev spec, then it will use that, otherwise it inits spec with an empty string. Maybe I should rename the function to `merge_previous_into_new_spec`?